packages/monaco/src/browser/monaco-command.ts

```diff
@@ -120,6 +126,57 @@ export class MonacoEditorCommandHandlers implements CommandContribution {
             }
         };
     }
+    protected newConfigTabSizeHandler(): MonacoEditorCommandHandler {
+        return {
+            execute: (editor: MonacoEditor) => {
+                const editorModel = editor.document;
+                if (editorModel && editorModel.textEditorModel) {
+                    const tabSize = editorModel.textEditorModel.getOptions().tabSize;
+                    const tabSizeOptions = Array.from(Array(8), (_, x) => x + 1)
+                        .map(size => new UpdateTabSizeOpenItem(size === tabSize ? `${size} *` : size.toString(), () => {
+                            editorModel.textEditorModel.updateOptions({
+                                tabSize: size,
+                                insertSpaces: editorModel.textEditorModel.getOptions().insertSpaces
+                            });
+                        })
+                        );
+                    this.open(tabSizeOptions, 'Change the Tab Size to...');
+                }
+            },
+            isEnabled: (editor: MonacoEditor) => !!editor
+        };
+    }
+    protected newConfigIndentationHandler(useSpaces: boolean): MonacoEditorCommandHandler {
+        return {
+            execute: (editor: MonacoEditor) => {
+                const editorModel = editor.document;
+                if (editorModel && editorModel.textEditorModel) {
+                    editorModel.textEditorModel.updateOptions({
+                        tabSize: editorModel.textEditorModel.getOptions().tabSize,
+                        insertSpaces: useSpaces
+                    });
+                }
+            },
+            isEnabled: (editor: MonacoEditor) => !!editor
+        };
+    }
+    private open(items: QuickOpenItem[], placeholder: string): void {
+        this.quickOpenService.open(this.getQuickOpenModel(items), this.getQuickOpenOptions(placeholder));
+    }
+    private getQuickOpenModel(items: QuickOpenItem[]): QuickOpenModel {
+        return {
+            onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+                acceptor(items);
+            }
+        };
+    }
+    private getQuickOpenOptions(placeholder: string, fuzzyMatchLabel: boolean = true): QuickOpenOptions {
+        return QuickOpenOptions.resolve({
```

Comment:
`QuickOpenOptions.resolve` is for the implementation, you can just return a json object